services/motion/builtin/move_attempt.go

```diff
@@ -0,0 +1,81 @@
+package builtin
+
+import (
+	"context"
+	"sync"
+
+	goutils "go.viam.com/utils"
+
+	"go.viam.com/rdk/utils"
+)
+
+// moveResponse is a struct that is used to communicate the outcome of a moveAttempt.
+type moveResponse struct {
+	err     error
+	success bool
+}
+
+// moveAttempt is a struct whose lifetime lasts the duration of an attempt to complete a moveRequest
+// it contains a context in which the move call executes and tracks the goroutines that it spawns.
+type moveAttempt struct {
+	ctx               context.Context
+	cancelFn          context.CancelFunc
+	backgroundWorkers *sync.WaitGroup
+
+	request      *moveRequest
+	responseChan chan moveResponse
+}
+
+// newMoveAttempt instantiates a moveAttempt which can later be started.
+// The caller of this function is expected to also call the cancel function to clean up after instantiation.
+func newMoveAttempt(ctx context.Context, request *moveRequest) *moveAttempt {
+	cancelCtx, cancelFn := context.WithCancel(ctx)
+	var backgroundWorkers sync.WaitGroup
+
+	return &moveAttempt{
+		ctx:               cancelCtx,
+		cancelFn:          cancelFn,
+		backgroundWorkers: &backgroundWorkers,
+
+		request:      request,
+		responseChan: make(chan moveResponse, 1),
```

Comment:
why is this buffered?

Reply:
I think if the first plan errors nothing will ever read from the channel as it tries to write the error

Reply:
but flushing the channel should fix that problem. Once tests pass, could we make it unbuffered & confirm that still works?

Reply:
Spoke offline and going to return an error in the start function